components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -361,7 +361,11 @@ bool CosmeticFiltersJSHandler::ProcessURL(
   enabled_1st_party_cf_ =
       force_cosmetic_filtering ||
       render_frame_->GetWebFrame()->IsCrossOriginToOutermostMainFrame() ||
-      content_settings->IsFirstPartyCosmeticFilteringEnabled(url_);
+      content_settings->IsFirstPartyCosmeticFilteringEnabled(url_) ||
+      net::registry_controlled_domains::SameDomainOrHost(
+          url_,
+          url::Origin::CreateFromNormalizedTuple("https", "youtube.com", 443),
```

Comment:
i would rather move this check to a separate statement, to avoid mixing exceptional/hardcoded cases with normal code. I.e. making corner cases very clear and visible never hurts

Reply:
+1 
Also ask iOS team to make a ticket to sync these changes.

Reply:
BTW we already use per-frame `GetBlinkPreferences().force_cosmetic_filtering`.
IMHO if we change `force_cosmetic_filtering` to `cosmetic_filtering_mode` we will get a single point to in the browser process to setup cosmetic filter mode for a single frame.

The exceptions like youtube & `g_vetted_search_engines` could be moved there.
And `cosmetic_filtering_mode` will be the one source to check for `CosmeticFiltersJsHandler`.

Reply:
I agree this check is not great, but I think these changes could probably move to a follow-up if done immediately following this since it's apparently a high priority fix, but my bigger concern here is that there is no test to verify this.

Reply:
The relevant iOS changes to correspond with these changes are here: It's introducing network level standard mode and the youtube.com exception to both network blocking and CF.